scripts/gen_api_vimdoc.py

```diff
@@ -471,67 +492,100 @@ def gen_docs(config):
                 if doc:
                     filename = os.path.basename(filename)
                     name = section_name.get(filename, name)
-                    title = '%s Functions' % name
-                    helptag = '*api-%s*' % name.lower()
+                    
+                    if filename.endswith('.lua'):
+                        title = '%s Lua Functions' % name
+                        helptag = '*lua-%s*' % name.lower()
+                    else:
+                        title = '%s Functions' % name
+                        helptag = '*api-%s*' % name.lower()
                     sections[filename] = (title, helptag, doc)
 
     if not sections:
         return
 
-    docs = ''
+    apidocs = ''
+    luadocs = ''
 
     i = 0
-    for filename in section_order:
+    for filename in api_section_order:
         if filename not in sections:
             continue
         title, helptag, section_doc = sections.pop(filename)
 
         i += 1
-        docs += sep
-        docs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
-        docs += section_doc
-        docs += '\n\n\n'
+        apidocs += sep
+        apidocs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
+        apidocs += section_doc
+        apidocs += '\n\n\n'
+
+    for filename in luadoc_section_order:
+        if filename not in sections:
+            continue
+        title, helptag, section_doc = sections.pop(filename)
+
+        i += 1
+        luadocs += sep
+        luadocs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
+        luadocs += section_doc
+        luadocs += '\n\n\n'
 
     if sections:
         # In case new API sources are added without updating the order dict.
         for title, helptag, section_doc in sections.values():
             i += 1
-            docs += sep
-            docs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
-            docs += section_doc
-            docs += '\n\n\n'
-
-    docs = docs.rstrip() + '\n\n'
-    docs += ' vim:tw=78:ts=8:ft=help:norl:\n'
-
-    doc_file = os.path.join(base_dir, 'runtime/doc', doc_filename)
-    delete_lines_below(doc_file, section_start_token)
-    with open(doc_file, 'ab') as fp:
-        fp.write(docs.encode('utf8'))
+            apidocs += sep
+            apidocs += '\n%s%s' % (title, helptag.rjust(text_width - len(title)))
+            apidocs += section_doc
+            apidocs += '\n\n\n'
+
+    apidocs = apidocs.rstrip() + '\n\n'
+    apidocs += ' vim:tw=78:ts=8:ft=help:norl:\n'
```

Comment:
Just pretend `apidocs` is "whatever docs we're currently working on". We don't want to have separate codepaths for every help doc we output.